sr_error_reporter/src/UnderactuationErrorReporter.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+* Copyright 2021 Shadow Robot Company Ltd.
+*
+* This program is free software: you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the Free
+* Software Foundation version 2 of the License.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "eigen_conversions/eigen_msg.h"
+#include "std_msgs/String.h"
+#include "sr_error_reporter/UnderactuationError.h"
+#include "sr_error_reporter/UnderactuationErrorReporter.hpp"
+
+UnderactuationErrorReporter::UnderactuationErrorReporter(ros::NodeHandle& node_handle)
+  : node_handle_(node_handle)
+{
+  joints_subscriber_ = node_handle.subscribe("/joint_states", 1,
+    &UnderactuationErrorReporter::joints_callback, this, ros::TransportHints().tcpNoDelay());
+  trajectory_subscriber_left_ = node_handle.subscribe("/lh_trajectory_controller/command", 1,
+    &UnderactuationErrorReporter::trajectory_callback, this, ros::TransportHints().tcpNoDelay());
+  trajectory_subscriber_right_ = node_handle.subscribe("/rh_trajectory_controller/command", 1,
+    &UnderactuationErrorReporter::trajectory_callback, this, ros::TransportHints().tcpNoDelay());
+
+  while (ros::ok())
+  {
+    std::string robot_description;
+    if (node_handle.searchParam("robot_description_semantic", robot_description))
+    {
+      robot_model_loader_.reset(new robot_model_loader::RobotModelLoader("robot_description"));
+      robot_state_.reset(new robot_state::RobotState(robot_model_loader_->getModel()));
+      break;
+    }
+    ROS_INFO("Waiting for robot description");
```

Comment:
ah we should wait for the parameter to be available actually, for reference look here, we have implemented a wait_for_param utility for cases like this one ->https://github.com/shadow-robot/jog_arm_internal/blob/f8dda4acd3ff02f006d2948fafb136a693b3bb4e/jog_arm/src/jog_arm/jog_arm_server.cpp#L98-L102

Reply:
Changed to use `wait_for_param` utility in bb8e6bc7b845fad190f9e568422db93d544fc863.